scripts/linters/pylint_extensions.py

```diff
@@ -1959,6 +2026,386 @@ def process_tokens(self, tokens):
+                        continue
+                    object_already_encountered_line_num = line_num
+                    # Excluding the case when object is called:
+                    # Eg: var = object()
+                    if 'object()' in line:
+                        continue
+                    if (
+                        object_comment_present and
+                        line_num <= (
+                            object_comment_line_num +
+                            self.ALLOWED_LINES_OF_GAP
+                        )
+                    ):
+                        object_comment_present = False
+                    else:
+                        self.add_message(
+                            'object-class-used', line=line_num)
+
+    def check_comment_is_present_with_cast_method(self, tokens):
+        """Checks whether the cast method in a module has been documented
+        or not. If the cast method is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        cast_type_regex = r'^# Here we use cast because'
+        cast_comment_present = False
+        inside_multi_line_import_scope = False
+        single_line_import = False
+        import_line_num = 0
+        cast_comment_line_num = 0
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            import_status_vars = [
+                single_line_import,
+                import_line_num,
+                inside_multi_line_import_scope
+            ]
+
+            if token_type == tokenize.COMMENT:
+                if re.search(cast_type_regex, line):
+                    cast_comment_present = True
+                    cast_comment_line_num = line_num
+
+            (
+                single_line_import,
+                import_line_num,
+                inside_multi_line_import_scope
+            ) = self._check_import_status(
+                import_status_vars,
+                token_type,
+                token,
+                line_num
+            )
+
+            if token_type == tokenize.NAME and token == 'cast':
+                # Passing those cases where cast is imported.
+                if single_line_import and import_line_num == line_num:
+                    pass
+                elif inside_multi_line_import_scope:
+                    pass
+                # Throwing an error when cast is encountered but there is no
+                # corresponding comment exist.
+                elif (
+                    cast_comment_present and
+                    line_num <= (
+                        cast_comment_line_num +
+                        self.ALLOWED_LINES_OF_GAP
+                    )
+                ):
+                    cast_comment_present = False
+                else:
+                    self.add_message(
+                        'cast-func-used', line=line_num)
```

Comment:
This is also pretty similar to this part of `Any`, please try to think how you can deduplicate the code where possible.

Reply:
The cast cannot be combined with the same helper function that we are using for Any and object, because:

1. We are not checking for the presence of cast in the function signature explicitly, whereas in Any and object we are checking for the function signature explicitly.
2. In Any and object we are excluding some cases but in cast, only the import case is excluded and for every other case we are throwing an error.
3. Logic of cast is a bit simple whereas the logic of object and Any is a bit complex as compared to cast. If we combine all together then it makes cast's logic complex too.